tools/changelog/src/update-latest-versions.js

```diff
@@ -0,0 +1,79 @@
+import { remark } from 'remark';
+import stringify from 'remark-stringify';
+import remarkParse from 'remark-parse';
+import fs from 'fs';
+import path from 'path';
+import { format } from 'date-fns';
+import { writeFile } from 'fs/promises';
+import getVersionHistory from './update-version-histories.js';
+
+const packagesDir = '../../packages';
+const versionOnlyRegex = /^\d+\.\d+\.\d+$/;
+
+function getAdaptorsFromDir() {
+  return fs
+    .readdirSync(packagesDir, { withFileTypes: true })
+    .filter(dirent => dirent.isDirectory())
+    .map(dirent => dirent.name);
+}
+
+async function updateChangelog(adaptorName, adaptorVersionList) {
+  const pkgPath = path.join(packagesDir, adaptorName);
+  const changelogPath = path.join(pkgPath, 'CHANGELOG.md');
+
+  if (!fs.existsSync(changelogPath)) return;
+
+  let markdownContent = fs.readFileSync(changelogPath, 'utf8');
+  const ast = await remark().use(remarkParse).parse(markdownContent);
+
+  const updatedAstChildren = ast.children.map((item, index) => {
+    if (item.type === 'heading' && item.depth === 2) {
```

Comment:
If we're not updating the whole version history, after checking the first header, we can exist this loop early. When updating the latest versions only, we literally only need to check the first header.